### PR TITLE
Adding skinning support in WireframeHelper

### PR DESCRIPTION
**Pull request by MiiBond**
This was mainly to allow for wireframe rendering of skinned `BufferGeometry` meshes (which isn't possible with the `material.wireframe` flag). Though, I also added regular `Geometry` support in `WireframeHelper` as well.

I also changed the generation of the wireframe for `BufferGeometry` so that only a new index buffer is generated (the other buffers can be shared with the mesh). This is much, much faster to create (the old way took several seconds for a large mesh). I'm not sure if there was a reason to regenerate the positions array for the wireframe. If so, I can change this back.

### CODE REVIEW

**Comment by MiiBond** [comment]
Hmm, I just noticed that I didn't implement the case for non-indexed `BufferGeometry`. I don't think I have an example model for this either. Does anyone know of a skinned, non-indexed, buffer-geometry model that I could use?

**Comment by WestLangley** [comment]
Nice! Would you consider showing an example use case? The easiest would be to modify `webgl_skinning_simple.html`.

**Comment by MiiBond** [comment]
Okay, I added it to `webgl_skinning_simple.html` though I didn't put it behind a toggle. Maybe I should?

**Comment by WestLangley** [comment]
That would be nice. It is up to you.

**Comment by MiiBond** [comment]
@WestLangley, I just realized that this was never accepted. I've updated it with the latest code and, unless there's something else, I think it's ready to go.

**Comment by WestLangley** [comment]
@MiiBond Does the demo work for you under r72dev?

**Comment by MiiBond** [comment]
I'm not near a computer right now but I'll try it later. It worked in our
framework with skinned meshes but I didn't actually try the example again.
Is it not working for you?

**Comment by mrdoob** [comment]
I'm thinking that maybe `WireframeGeometry` could do

```javascript
if ( geometry instanceof THREE.Geometry ) {

    geometry = new THREE.BufferGeometry().fromGeometry( geometry );

}
```

and simplify the code.

**Comment by mrdoob** [comment]
Or as `EdgesGeometry` already does...

```javascript
if ( geometry instanceof THREE.BufferGeometry ) {

    geometry = new THREE.Geometry().fromBufferGeometry( geometry );

}
```

**Comment by MiiBond** [comment]
@WestLangley Just tried this out this morning. Yeah, that 'simple' model doesn't animate properly anymore. The marine still works though. In both cases, the wireframe matches the skinning of the model so I think WireframeHelper is working as it should.

Another issue is that the example was still using MeshFaceMaterial which was removed. I've changed it to MeshLambertMaterial for now. Maybe we can change this back when multi-material meshes are restored?

File: src/extras/helpers/WireframeHelper.js [hunk 1]
```diff
@@ -6,7 +6,14 @@ THREE.WireframeHelper = function ( object, hex ) {
...
 
-	THREE.LineSegments.call( this, new THREE.WireframeGeometry( object.geometry ), new THREE.LineBasicMaterial( { color: color } ) );
+	var geometry = new THREE.WireframeGeometry( object.geometry );
+	var skinning = !!geometry.skinning;
+	if ( skinning && object.skeleton ) {
+		this.skeleton = object.skeleton;
+		this.bindMatrix = object.bindMatrix;
+		this.bindMatrixInverse = object.bindMatrixInverse;
+	}
+	THREE.LineSegments.call( this, new THREE.WireframeGeometry( object.geometry ), new THREE.LineBasicMaterial( { color: color, skinning: skinning } ) );
```
Comment (WestLangley) [inline src/extras/helpers/WireframeHelper.js]: THREE.LineSegments.call( this, geometry, new THREE.LineBasicMaterial( { color: color, skinning: skinning } ) );

**Comment by WestLangley** [comment]
Yes, you left out skinning in the non-indexed `BufferGeometry` case. Do you have a test case for both of the cases that you did implement?

**Comment by MiiBond** [comment]
I'm working on that right now along with using the ```fromGeometry``` suggestion from @mrdoob. I'm in meetings all morning though so it might not be until the afternoon or tomorrow before I get this done.

**Comment by WestLangley** [comment]
Hmmm. I am not sure I would remove the `BufferGeometry` code and replace it with

    if ( geometry instanceof THREE.BufferGeometry ) {

        geometry = new THREE.Geometry().fromBufferGeometry( geometry );

   }

Someday, `THREE.Geometry` may be removed. No?

**Comment by MiiBond** [comment]
No, I was imagining going with @mrdoob 's first comment. i.e. 
```if ( geometry instanceof THREE.Geometry ) {
    geometry = new THREE.BufferGeometry().fromGeometry( geometry );
}```

**Comment by WestLangley** [comment]
FWIW, if I remember correctly, I think the `EdgesGeometry` code _requires_ the `THREE.Geometry` data structure.

**Comment by MiiBond** [comment]
Okay, I finally got around to doing this. It should be good to go now for ```THREE.Geometry``` and indexed and non-indexed ```THREE.BufferGeometry```.

**Comment by mrdoob** [comment]
> Someday, `THREE.Geometry` may be removed. No?

I tried. But I think it's going to stay around for a while... 😅

File: src/extras/helpers/WireframeHelper.js [hunk 2]
```diff
@@ -6,7 +6,14 @@ THREE.WireframeHelper = function ( object, hex ) {
 
 	var color = ( hex !== undefined ) ? hex : 0xffffff;
 
-	THREE.LineSegments.call( this, new THREE.WireframeGeometry( object.geometry ), new THREE.LineBasicMaterial( { color: color } ) );
+	var geometry = new THREE.WireframeGeometry( object.geometry );
+	var skinning = !!geometry.skinning;
```
Comment (mrdoob) [inline src/extras/helpers/WireframeHelper.js]: `geometry.skinning === true`? ^^

**Comment by MiiBond** [comment]
Done.
I also switched the example back to use the 'simple.js' model. It doesn't seem to be working correctly right now but that appears to be an unrelated issue.

**Comment by MiiBond** [comment]
Is it possible to merge this in or do we want to make more changes?

**Comment by WestLangley** [comment]
@MiiBond +1, but can you simplify this PR so it is more easily understood? There should be only a handful of files changed.

**Comment by MiiBond** [comment]
Created a new pull request at #6957

**Pull request closed without merging**